src/main/java/uk/gov/hmcts/divorce/common/event/Applicant2SolicitorApplyForFinalOrder.java

```diff
@@ -0,0 +1,239 @@
+
+    public static final String FINAL_ORDER_REQUESTED_APP2_SOL = "final-order-requested-app2-sol";
+
+    public static final String APPLY_FOR_FINAL_ORDER = "Apply for final order";
+
+    @Autowired
+    private Clock clock;
+
+    @Autowired
+    private Applicant2AppliedForFinalOrderNotification applicant2AppliedForFinalOrderNotification;
+
+    @Autowired
+    private FinalOrderRequestedNotification finalOrderRequestedNotification;
+
+    @Autowired
+    private NotificationDispatcher notificationDispatcher;
+
+    @Autowired
+    private ApplyForFinalOrderService applyForFinalOrderService;
+
+    @Autowired
+    private GeneralReferralService generalReferralService;
+
+    @Autowired
+    private PaymentService paymentService;
+
+    @Autowired
+    private SolFinalOrderPayment solFinalOrderPayment;
+
+    @Override
+    public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        final List<CcdPageConfiguration> pages = List.of(
+            new Applicant2SolApplyForFinalOrderDetails(),
+            new Applicant2SolFinalOrderExplainWhyNeedToApply(),
+            solFinalOrderPayment,
+            new HelpWithFeesPageForApplicant2SolFinalOrder(),
+            new SolFinalOrderPayAccount(),
+            new SolFinalOrderPaymentSummary()
+        );
+
+        final PageBuilder pageBuilder = addEventConfig(configBuilder);
+        pages.forEach(page -> page.addTo(pageBuilder));
+    }
+
+    private PageBuilder addEventConfig(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        return new PageBuilder(configBuilder
+            .event(FINAL_ORDER_REQUESTED_APP2_SOL)
+            .forStates(AwaitingFinalOrder)
+            .name(APPLY_FOR_FINAL_ORDER)
+            .description(APPLY_FOR_FINAL_ORDER)
+            .showCondition("applicationType=\"soleApplication\" AND finalOrderReminderSentApplicant2=\"Yes\"")
+            .showSummary()
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE, APPLICANT_2_SOLICITOR)
+            .aboutToStartCallback(this::aboutToStart)
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grantHistoryOnly(
+                CASE_WORKER,
+                SUPER_USER,
+                LEGAL_ADVISOR,
+                APPLICANT_1_SOLICITOR,
+                JUDGE));
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
+        log.info("{} about to start callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        CaseData data = details.getData();
+        log.info("Retrieving order summary");
+        final OrderSummary orderSummary = paymentService.getOrderSummaryByServiceEvent(SERVICE_OTHER, EVENT_GENERAL, KEYWORD_NOTICE);
+
+        data.getFinalOrder().setApplicant2SolFinalOrderFeeOrderSummary(orderSummary);
```

Comment:
Question: If we're populating the order summary in the aboutToStart, what happens if the user triggers the aboutToStart but cancels out of the event? Does it set the order summary / display it somewhere?